oceannavigator/frontend/src/components/Map/utils.js

```diff
@@ -636,6 +672,57 @@ export const createFeatureVectorLayer = (source, mapSettings) => {
   });
 };
 
+export const createFeatureTextStyle = (
+  feature,
+  textColor = "#000",
+  strokeColor = "#ffffff",
+  mapSettings = null
+) => {
+  if (!feature.get("name")) return null;
+
+  const type = feature.get("type");
+  let geometry = undefined;
+  let textBaseline = "middle";
+  let offsetY = 0;
+
+  if (type === "LineString") {
+    // For LineString
+    textBaseline = "bottom";
+    offsetY = -8;
+  } else if (type === "Polygon") {
+    // Special case for predefined polygons
+    if (feature.get("key") && mapSettings) {
+      geometry = new olgeom.Point(
+        olProj.transform(
+          feature.get("centroid"),
+          "EPSG:4326",
+          mapSettings.projection
+        )
+      );
+    } else {
+      // Regular polygons
+      let centroid = feature.get("labelCentroid");
+      if (!centroid) {
+        centroid = feature.getGeometry().getInteriorPoint().getCoordinates();
+        feature.set("labelCentroid", centroid);
+      }
+      geometry = new olgeom.Point(centroid);
+    }
```

Comment:
You could easily reduce this by adding a `centroid` property to the feature instead of `labelCentroid`.  Something like this could work:

```suggestion
  } else if (type === "Polygon") {
    if (!feature.get("centroid")) {
      let centroid = feature.getGeometry().getInteriorPoint().getCoordinates();
      feature.set("centroid", olProj.toLonLat(centroid).slice(0, 2));
    }
    geometry = new olgeom.Point(
      olProj.transform(
        feature.get("centroid"),
        "EPSG:4326",
        mapSettings.projection
      )
    );
    }
```